src/stores/nav.js

```diff
@@ -0,0 +1,28 @@
+import { defineStore } from 'pinia'
+import { reactive, toRefs } from '@nuxtjs/composition-api'
+
+export const NAV = 'nav'
+
+/**
+ * Store information about navigation.
+ */
+export const useNavStore = defineStore(NAV, () => {
+  const state = reactive({
+    isEmbedded: true,
+    isReferredFromCc: false,
+  })
+  const { isEmbedded, isReferredFromCc } = toRefs(state)
+
+  function setIsEmbedded(isEmbedded = true) {
+    state.isEmbedded = isEmbedded
+  }
+  function setIsReferredFromCc(isReferredFromCc = true) {
+    state.isReferredFromCc = isReferredFromCc
+  }
+  return {
+    isEmbedded,
+    isReferredFromCc,
+    setIsEmbedded,
+    setIsReferredFromCc,
+  }
+})
```

Comment:
It looks like we use a mix of updating the `isEmbedded` and `isReferredFromCc` values directly and also through the setter functions. Should we standardize on one approach for simple properties like this?

https://github.com/WordPress/openverse-frontend/blob/7c59473bdc4e0aaa86d0c5b9174a90808e914249/src/middleware/middleware.js#L33

https://github.com/WordPress/openverse-frontend/blob/7c59473bdc4e0aaa86d0c5b9174a90808e914249/src/plugins/migration-notice.js#L11

Reply:
I prefer setter functions because they are easier to observe.

Reply:
If we went with setter functions is there a way to enforce their usage, like to make `isReferredFromCc` readonly?

Reply:
Yes! I'll do that